cmd/mesos_metrics_check.go

```diff
@@ -0,0 +1,156 @@
+// Copyright © 2017 Mesosphere Inc. <http://mesosphere.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/dcos/dcos-checks/client"
+	"github.com/dcos/dcos-go/dcos"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const (
+	mesosMasterHTTPPort = 5050
+	mesosAgentHTTPPort  = 5051
+	nodeRecovered       = 1.0
+)
+
+// mesosMetricsCmd represents the mesos-metrics command
+var mesosMetricsCmd = &cobra.Command{
+	Use:   "mesos-metrics",
+	Short: "Get the mesos metrics snapshot",
+	Long:  `Metrics snapshot lets us know if the mesos rep logs are synchronized`,
+	Run: func(cmd *cobra.Command, args []string) {
+		RunCheck(context.TODO(), NewMesosMetricsCheck("DC/OS metrics snapshot check"))
+	},
+}
+
+// NewMesosMetricsCheck returns an initialized instance of *MesosMetricsCheck.
+func NewMesosMetricsCheck(name string) DCOSChecker {
+	check := &MesosMetricsCheck{Name: name}
+	check.urlFunc = check.getURL
+	return check
+}
+
+// MesosMetricsCheck struct
+type MesosMetricsCheck struct {
+	Name    string
+	urlFunc func(*http.Client, *CLIConfigFlags) (*url.URL, error)
+}
+
+// ID returns a unique check identifier.
+func (mm *MesosMetricsCheck) ID() string {
+	return mm.Name
+}
+
+type masterResponse struct {
```

Comment:
`masterResponse` and `agentResponse` are declared at the package level but are likely only used in this particular check.  There's probably not much reason to leave them at the package level.  These should probably be moved into the `Run` method.